service/src/test/java/uk/gov/hmcts/reform/fpl/service/removeorder/UpdateCMOHearingTest.java

```diff
@@ -0,0 +1,129 @@
+            .type(DRAFT_CMO)
+            .status(DRAFT)
+            .hearing(hearingWithCMOId.toLabel())
+            .build());
+
+        HearingOrdersBundle hearingOrdersBundle = HearingOrdersBundle.builder()
+            .hearingId(HEARING_ID)
+            .orders(newArrayList(linkedCMO,
+                element(ANOTHER_CMO_ID, HearingOrder.builder().build())))
+            .build();
+
+        CaseData caseData = CaseData.builder()
+            .hearingDetails(List.of(element(HEARING_ID, hearingWithCMOId),
+                element(ANOTHER_HEARING_ID, hearing(randomUUID()))))
+            .hearingOrdersBundlesDrafts(List.of(
+                element(HEARING_BUNDLE_ID, hearingOrdersBundle)))
+            .build();
+
+        assertThat(underTest.getHearingToUnlink(caseData, CMO_ID, linkedCMO.getValue()))
+            .isEqualTo(hearingWithCMOId);
+    }
+
+    @Test
+    void shouldThrowExceptionWhenNoHearingFoundForTheCMOId() {
+        HearingBooking hearingWithCMOId = hearing(CMO_ID, HEARING_START_DATE);
+
+        Element<HearingOrder> sealedCMO = element(CMO_ID, HearingOrder.builder()
+            .type(AGREED_CMO).status(APPROVED)
+            .hearing(hearingWithCMOId.toLabel()).build());
+
+        HearingOrder anotherCMO = HearingOrder.builder().type(AGREED_CMO).status(APPROVED).build();
+
+        CaseData caseData = CaseData.builder()
+            .sealedCMOs(newArrayList(sealedCMO))
+            .hearingDetails(List.of(element(HEARING_ID, hearing(CMO_ID))))
+            .build();
+
+        Exception exception = assertThrows(UnexpectedNumberOfCMOsRemovedException.class,
+            () -> underTest.getHearingToUnlink(caseData, ANOTHER_CMO_ID, anotherCMO));
+
+        AssertionsForClassTypes.assertThat(exception).hasMessageContaining(String.format(
+            "CMO %s could not be linked to hearing by CMO id and there wasn't a unique link "
+                + "(%s links found) to a hearing with the same label", ANOTHER_CMO_ID, 0));
+    }
+
+    @Test
+    void shouldRemoveHearingLinkedToCMO() {
+        LocalDateTime differentStartDate = HEARING_START_DATE.plusDays(3);
+        HearingBooking hearingWithCMOId = hearing(CMO_ID, differentStartDate);
+        HearingBooking anotherHearing = hearing(ANOTHER_CMO_ID, HEARING_START_DATE.plusDays(4));
+
+        Element<HearingOrder> linkedCMO = element(CMO_ID, HearingOrder.builder()
+            .type(DRAFT_CMO)
+            .status(DRAFT)
+            .hearing(hearingWithCMOId.toLabel())
+            .build());
+
+        HearingOrdersBundle hearingOrdersBundle = HearingOrdersBundle.builder()
+            .hearingId(HEARING_ID)
+            .orders(newArrayList(linkedCMO,
+                element(ANOTHER_CMO_ID, HearingOrder.builder().build())))
+            .build();
+
+        CaseData caseData = CaseData.builder()
+            .hearingDetails(List.of(element(HEARING_ID, hearingWithCMOId),
+                element(ANOTHER_HEARING_ID, anotherHearing)))
```

Comment:
minor, list.of(element(x),element(y) can be replaced like..

           .hearingDetails(wrapElements(HEARING_ID, hearingWithCMOId),
                (ANOTHER_HEARING_ID, anotherHearing)

Reply:
discussed and leaving as not currently working like this